migration-0.38/migration/alerts.py

```diff
@@ -0,0 +1,105 @@
+import sqlite3
+import json
+from clickhouse_driver import Client
+import sys
+import json
+from migration.fields import update_field
+
+
+
+def update_alert(data, fields):
+
+    query_type = data['condition']['compositeQuery']['queryType']
+    if query_type == "clickhouse_sql":
+        # print(testing)
+        # there will only be one query for ch in alerts
+        clickhouse_sql = data['condition']['compositeQuery']['chQueries']['A']
+        if "query" not in clickhouse_sql.keys():
+            print("Query not found : {} , type: {}".format(data['alert'], query_type))
+            print(clickhouse_sql)
+            return data
+        clickhouse_sql =clickhouse_sql["query"]
+        
+        if "signoz_logs.distributed_logs"  in clickhouse_sql:
+            print("Alert : {} , type: {}".format(data['alert'], query_type))
+            for old_name, updated_attribute in fields.items():
+                # check if attribute is there
+                # check is done by checking # attributes_string_key, 'request_context_test_name'
+                check = updated_attribute[2] + "_" + updated_attribute[1].lower() + "_key, '" + updated_attribute[0].replace('.', '_') + "'"
+                if check in clickhouse_sql:
+                    updated = updated_attribute[2] + "_" + updated_attribute[1].lower() + "_key, '" + updated_attribute[0] + "'"
+                    clickhouse_sql = clickhouse_sql.replace(check, updated)
+
+                
+                # check if materialized column is there if yes then replace it
+                materialized_name = updated_attribute[2][:-1] + "_" + updated_attribute[1].lower() + "_" + updated_attribute[0].replace('.', '_')
+                if materialized_name in clickhouse_sql:
+                    updated_materialized_name = updated_attribute[2][:-1] + "_" + updated_attribute[1].lower() + "_" + updated_attribute[0].replace('.', '$$')
+                    clickhouse_sql = clickhouse_sql.replace(materialized_name, updated_materialized_name)
+                data['condition']['compositeQuery']['chQueries']['A']['query'] = clickhouse_sql 
+                      
+    elif query_type == "builder":
+        groupByNames = {}
+        for name, builderQuery in data['condition']['compositeQuery']['builderQueries'].items():
+            data_source = builderQuery.get('dataSource', 'No DataSource found')
+
+            ## dont allow alerts which 
+            if data_source != "logs" and name == builderQuery["expression"]:
+                break
```

Comment:
Should you be continuing or breaking here? Say if there was a traces query first and logs second, Ex: the author is trying to compare something from traces with something else in logs, The first query data source is "traces" nad `name=A` and `builderQuery["expression"] =A` as well.

Reply:
thanks, you are correct.